### PR TITLE
Fix sort

### PR DESCRIPTION
**Pull request by MBkkt**

### CODE REVIEW

**Comment by Dronplane** [comment]
Are there any tests?

**Review by Dronplane — APPROVED** [review]
LGTM but I guess codecs should be the same in test see lines 614 and 599 in segment_writer_tests.cpp

File: tests/index/segment_writer_tests.cpp [hunk 1]
```diff
@@ -518,3 +521,106 @@ TEST_F(segment_writer_tests, index_field) {
...
+             std::span<irs::segment_writer::update_context> ctxs,
+             std::vector<size_t> order) {
+  for (size_t i = 0; i < order.size(); ++i) {
+    auto new_i = order[i];
+    while (i != new_i) {
+      std::swap(docs[i], docs[new_i]);
+      std::swap(ctxs[i], ctxs[new_i]);
+      auto new_new_i = order[new_i];
+      order[new_i] = new_i;
+      new_i = new_new_i;
```
Comment (gnusi) [inline tests/index/segment_writer_tests.cpp:552]:
```suggestion
std::swap(new_i, order[new_i]);
```

File: tests/index/sorted_index_tests.cpp [hunk 2]
```diff
@@ -1951,4 +1953,124 @@ INSTANTIATE_TEST_SUITE_P(
...
+  std::array<std::pair<size_t, std::unique_ptr<irs::by_term>>, kLen>
+    remove_docs;
+  for (size_t i = 0; i < kLen; ++i) {
+    remove_docs[i] = {i, MakeByTerm("name", static_cast<tests::string_field&>(
+                                              *insert_docs[i].second->sorted)
+                                              .value())};
+  }
+  std::array<bool, kLen> in_store;
+  std::array<char, kLen> results{'A', 'B', 'C', 'D', 'E'};
+  for (size_t reset = 0; reset < 64; ++reset) {
```
Comment (MBkkt) [inline tests/index/sorted_index_tests.cpp:1987]:
```suggestion
  for (size_t reset = 0; reset < 32; ++reset) {
```

**Pull request merged**